File: apps/desktop/src/types/asset_version.ts
export interface AssetVersion {
  id: string;
  entity_type: string;
  entity_id: string;
  version: number;
  model_id: string;
  provider_id: string;
  job_id: string;
  payload: Record<string, unknown>;
  is_current: boolean;
  created_at: string;
  file_url: string;
}

File: apps/backend/app/schemas/asset_version.py
"""Phase 9 — 资产版本 Schema。"""

from datetime import datetime

from pydantic import BaseModel, Field


class AssetVersionOut(BaseModel):
    id: str
    entity_type: str
    entity_id: str
    version: int
    model_id: str = ""
    provider_id: str = ""
    job_id: str = ""
    payload: dict = Field(default_factory=dict)
    is_current: bool
    created_at: datetime
    file_url: str
